hangman.rb

```diff
@@ -0,0 +1,202 @@
+# it's not a fully implemented feature: it doesn't add a piece to the hangman for
+# bad guesses. But it does let you skip ahead to winning if you already know the
+# word, which is what I was going for. It also throws some errors. >_>
+
+# -*- coding: utf-8 -*-
+require "colorize"
+
+# pair project w/ Wang :)
+
+class Hangman
+# 1. {0}
+# 2. (\
+# 3. /)
+# 4. ~
+# 5. (:)
+# 6. _/
+# 7. \_
+
+  def initialize
+    # hangman board:
+    @first        = "     ______       "
+    @second       = "    |      |      "
+    @third        = "           |      "
+    @fourth       = "           |      "
+    @fifth        = "           |      "
+    @sixth        = "           |      "
+    @seventh      = "    _______|___   "
+    @eighth       = "   |           |  "
+    @ninth        = "                  "
+    @tenth        = "  H A N G M A N   " # change based on how many letters, etc
+
+    # start game
+    start
+  end
+
+
+  def display_board
+    @hangman = [@first, @second, @third, @fourth, @fifth,
+                @sixth, @seventh, @eighth, @ninth, @tenth]
+
+
+    @hangman.each do |line|
+      puts line
+    end
+
+    puts "\n"
+  end
+
+
+  def set_answer
+    possible_answers = [ "potato", "broccoli", "animal", "hotdog",
+      "elephant", "cat", "taco", "whale", "walrus", "vulture", "lunch",
+      "morning", "evening", "scotch", "whiskey", "salad", "notebook",
+      "laptop", "coffee", "caffeine", "solar", "system", "planet",
+      "cheerful", "happy", "emotion", "below", "scream", "shout", "jumprope",
+      "wizard", "rainfall", "cloudy", "sunlight", "sunshine", "program",
+      "hangman", "sticky", "oranges", "eraser", "pencil", "powerful",
+      "creative", "cleaning", "nontoxic", "keyboard", "key", "any", "four",
+      "badge", "ridge", "cringe", "fringe", "lawful", "lawless", "chaotic",
+      "chaos", "crafty", "space", "outside", "outdoor", "cabin", "camping",
+      "tentacle", "monster", "you", "our", "hour"
+      ]
+
+    # select random word
+    which_random_index = rand(0...possible_answers.length)
+    @answer = possible_answers[which_random_index]
+
+    # update hangman word display
+    @tenth = " _" * @answer.length
+  end
+
+
+  def start
+    puts "Let's play hangman!"
+
+    @now_playing = true
+    @bad_count = 0
+    @good_count = 0
+    @letters_guessed = []
+
+    set_answer
+    display_board
+    play
+  end
+
+
+  def play
+    while @now_playing
+      guess = get_guess
+      check_guess(guess)
+
+      display_board
+    end
+  end
+
+
+  def get_guess
+    puts "Guess your letter:"
+
+    input = gets.chomp.downcase # downcase == shouldn't matter if cap letters
+
+    if input == "quit"
+      exit
+    end
+
+    if input == @answer # this isn't finished; it doesn't increment @bad_count
+      return win        # or call bad_guess.
+    end
+
+    until (input >= "a") && (input <= "z") && (input.length == 1) # while these aren't true / until they are
```

Comment:
This is great! You can also use a letter range `("a".."z").include?` to check if the letter is included within the range.